src/target/riscv/riscv.c

```diff
@@ -4718,6 +4718,10 @@ unsigned int riscv_count_harts(struct target *target)
  */
 static bool gdb_regno_cacheable(enum gdb_regno regno, bool is_write)
 {
+	/* Don't cache values of x0 -- this prevents storing non-zero (bogus) values in the cache. */
```

Comment:
There seems to be a miss-match between the comment and the action -- currently, `x0` is cached on reads.
I would suggest to delete the comment, since, IMHO, the code is self-explanatory.